nfsn-setup.sh

```diff
@@ -9,27 +9,25 @@ git submodule init
 git submodule update --remote
 mkdir -p letsencrypt.sh/.acme-challenges
 
+# write_config $config_dir $wellknown
+readonly hook_path=$(realpath nfsn-hook.sh)
+write_config() {
+    mkdir -p -- "$1" "$2"
+    printf '%s=%q\n' WELLKNOWN "$2" HOOK "$hook_path" >"$1/config"
+}
+
 echo " + Generating configuration..."
 for site_root in $(nfsn list-aliases); do
    if [[ -d "${DOCUMENT_ROOT}${site_root}/" ]]; then
-      WELLKNOWN="${DOCUMENT_ROOT}${site_root}/${well_known}"
-      CONFIGDIR="letsencrypt.sh/certs/${site_root}/"
-      mkdir -p "${WELLKNOWN}" "${CONFIGDIR}"
-      echo "WELLKNOWN='${WELLKNOWN}'" > "${CONFIGDIR}/config"
-      echo " + Installing hook script..."
-      echo "HOOK='$(realpath nfsn-hook.sh)'" >> "${CONFIGDIR}/config"
-      chmod +x nfsn-hook.sh
+      write_config "letsencrypt.sh/certs/${site_root}/" \
```

Comment:
One final nitpick: The trailing slash here results in a double-slash within the function as you write to `$1/config`, thus `${site_root}//config`.